src/crypto/crypto.h

```diff
@@ -162,6 +163,16 @@ namespace crypto {
     return res;
   }
 
+  /* Generate a random value between range_min and range_max
+   */
+  template<typename T>
+  typename std::enable_if<std::is_integral<T>::value, T>::type rand_range(T range_min, T range_max) {
+    std::random_device rd;
```

Comment:
We might want to define our own random generator to keep the behavior the same. The `mt19937` isn't a crypto-secure generator:
```c++
  struct random_device
  {
    typedef uint64_t result_type;
    static constexpr result_type min() { return 0; }
    static constexpr result_type max() { return result_type(-1); }
    result_type operator()() const { return crypto::rand<result_type>(); }
  };
```
will work directly with `uniform_int_distribution` and doesn't do a syscall each time (similar to `mt133937` usage).